common/men_at_arms_types/00_maa_types.txt

```diff
@@ -423,6 +423,33 @@ massive_aerial_unit = {
 	
 }
 
+riflemen = {
+	type = archers
+	
+	damage = 60
+	toughness = 10
+	pursuit = 10
+	screen = 0
+	
+	terrain_bonus = {
+		jungle = { damage = -15 }
+		forest = { damage = -10 }
+	}
+
+	counters = {
+		heavy_infantry = 1
+		heavy_cavalry = 1
+		massive_aerial = 1
+	}
+
+	buy_cost = { gold = riflemen_recruitment_cost }
```

Comment:
So how you made sure the AI picking riflemen over archers and crossbowmen?

Reply:
why do some units not have any ai_quality value?  I could add more value to archers, but im not sure how crossbowmen get a boost because they have no ai boost 
![image](https://user-images.githubusercontent.com/74790101/120870341-a9bcf080-c566-11eb-8bbb-43fc4cbcf36d.png)

Reply:
> why do some units not have any ai_quality value?

No one touched it.

> I could make the separate unit type, if we dont figure it out

Figure what out?

Reply:
That was an image from vanilla

So for the unit type archers we have:

bowmen(basic)
and 
crossbowmen

the bowmen use an ai_quality score (  ai_quality = { value = culture_ai_weight_archers }  ) while the sub-unit crossbowmen does not, yet the ai picks it anyway somehow and idk how. I think doing:
ai_quality = { value = culture_ai_weight_archers +20 } for the riflemen could work, but again idk how this system works

btw siege weapons dont use these values at all and are always picked by ai

Reply:
This is what I cant figure out^

Reply:
> the bowmen use an ai_quality score ( ai_quality = { value = culture_ai_weight_archers } ) while the sub-unit crossbowmen does not, yet the ai picks it anyway somehow and idk how.

It provides better attack and defense for its money than the archers.

Reply:
See `TOUGHNESS_SCORE_MULT`, etc. in `00_ai.txt`

Reply:
That I get,  what I dont get is why they dont use the ai_quality, riflemen have better stats than bowmen so if I dont need to add an ai_quality value then I guess it's been resolved.

Reply:
> I dont get is why they dont use the ai_quality
Paradox didn't need it because the AI picked it anyway, ez.

Reply:
![image](https://user-images.githubusercontent.com/74790101/121757956-75f54400-caed-11eb-9e46-4c4fba1c5b2f.png)
![image](https://user-images.githubusercontent.com/74790101/121757961-7d1c5200-caed-11eb-8e4e-a480a6791cab.png)
Ok I see what you mean, so if I change the stats to this we are all set?  It would be a better buy than both bowmen and crossbowmen and only slightly underpriced.

Reply:
> Ok I see what you mean, so if I change the stats to this we are all set? It would be a better buy than both bowmen and crossbowmen and has the balance factor close too the other two.

Yes, the riflemen have a higher **_AI Score_** so the AI will buy it.